python/cuml/feature_extraction/text.py

```diff
@@ -0,0 +1,473 @@
+    binary : boolean, default=False
+        If True, all non zero counts are set to 1. This is useful for discrete
+        probabilistic models that model binary events rather than integer
+        counts.
+    dtype : type, optional
+        Type of the matrix returned by fit_transform() or transform().
+    remove_punctuation : boolean, True by default
+        Remove all characters from string.punctuation before tokenizing.
+    Attributes
+    ----------
+    vocabulary_ : nvstrings
+        Array mapping from feature integer indices to feature name.
+    stop_words_ : nvstrings
+        Terms that were ignored because they either:
+          - occurred in too many documents (`max_df`)
+          - occurred in too few documents (`min_df`)
+          - were cut off by feature selection (`max_features`).
+        This is only available if no vocabulary was given.
+    """
+    def __init__(self, input=None, encoding=None, decode_error=None,
+                 strip_accents=None, lowercase=True, preprocessor=None,
+                 tokenizer=None, stop_words=None, token_pattern=None,
+                 ngram_range=(1, 1), analyzer='word', max_df=1.0, min_df=1,
+                 max_features=None, vocabulary=None, binary=False,
+                 dtype=cp.int32, remove_punctuation=True):
+        self.preprocessor = preprocessor
+        self.analyzer = analyzer
+        self.lowercase = lowercase
+        self.remove_punctuation = remove_punctuation
+        self.stop_words = stop_words
+        self.max_df = max_df
+        self.min_df = min_df
+        if max_df < 0 or min_df < 0:
+            raise ValueError("negative value for max_df or min_df")
+        self.max_features = max_features
+        if max_features is not None:
+            if not isinstance(max_features, int) or max_features <= 0:
+                raise ValueError(
+                    "max_features=%r, neither a positive integer nor None"
+                    % max_features)
+        self.ngram_range = ngram_range
+        self.vocabulary = vocabulary
+        self.binary = binary
+        self.dtype = dtype
+
+        sklearn_params = {"input": input,
+                          "encoding": encoding,
+                          "decode_error": decode_error,
+                          "strip_accents": strip_accents,
+                          "tokenizer": tokenizer,
+                          "token_pattern": token_pattern}
+        self.check_sklearn_params(analyzer, sklearn_params)
+
+    def _count_vocab(self, docs):
+        """Create feature matrix, and vocabulary where fixed_vocab=False"""
+        vocabulary = self.vocabulary_
+        X = cp.empty((len(docs), len(vocabulary)), dtype=cp.int32)
+
+        if self.ngram_range != (1, 1):
+            docs = self._surround_with_space(docs)
+            vocabulary = self._surround_with_space(vocabulary)
+            nvtext.strings_counts(docs, vocabulary, devptr=X.data.ptr)
```

Comment:
You might not be able to use `nvtext.strings_counts` as that relies on a dense vector as input. Might have to treat this as series and solve this via something like below:

https://github.com/rapidsai/tpcx-bb/blob/02776c48fb6ac7580327d5626f109a6fb6cd5308/tpcx_bb/xbb_tools/text_vectorizers/dist_hashing_vectorizer.py#L148-L190